### PR TITLE
SUP-1312: Pipeline Template resource implementation

### PR DESCRIPTION
**Pull request by james2791**
feat(Pipeline Template resource): Added support for pipeline template resource CRUD/import ops

## PR checklist:
- [x] `docs/` updated
- [x] tests added
- [x] an example added to `examples/` (useful to demo new field/resource)
- [x] `CHANGELOG.md` updated with pending release information

### CODE REVIEW

File: buildkite/graphql/node.graphql [hunk 1]
```diff
@@ -5,6 +5,9 @@ query getNode(
         ... on Pipeline {
             ...PipelineValues
         }
+        ... on PipelineTemplate {
+            ...PipelineTemplateValues
+        }
         ... on TeamMember {
             ...TeamMemberValues
         }
```
Comment (mcncl) [inline buildkite/graphql/node.graphql:13]: Consistency on `fields` or `values` might be worth considering.

File: buildkite/provider.go [hunk 2]
```diff
@@ -139,6 +139,7 @@ func (tf *terraformProvider) Resources(context.Context) []func() resource.Resour
 		newTeamResource,
 		newTestSuiteResource,
 		newPipelineTeamResource,
+		NewPipelineTemplateResource,
```
Comment (mcncl) [inline buildkite/provider.go:142]: Should be lower case; it is internal to the package.

File: buildkite/resource_pipeline_template.go [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+	Configuration types.String `tfsdk:"configuration"`
+	Description   types.String `tfsdk:"description"`
+	Name          types.String `tfsdk:"name"`
+}
+
+type PipelineTemplateResource struct {
+	client *Client
+}
+
+func NewPipelineTemplateResource() resource.Resource {
```
Comment (mcncl) [inline buildkite/resource_pipeline_template.go:32]: Consider casing throughout.

File: buildkite/graphql/node.graphql (hunk 1 above)
Comment (james2791) [reply buildkite/graphql/node.graphql]: Im for the former - I can head the renaming of the other two Fragments in another PR :)

File: buildkite/provider.go (hunk 2 above)
Comment (james2791) [reply buildkite/provider.go]: Tweaked - case of existing resources too, which I think we can consolidateA

File: buildkite/resource_pipeline_template.go (hunk 3 above)
Comment (james2791) [reply buildkite/resource_pipeline_template.go]: Tweaked

File: buildkite/resource_pipeline_template.go [hunk 4]
```diff
@@ -0,0 +1,302 @@
...
+	diagsPlan := req.Plan.Get(ctx, &plan)
+
+	resp.Diagnostics.Append(diagsState...)
+	resp.Diagnostics.Append(diagsPlan...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	timeout, diags := pt.client.timeouts.Create(ctx, DefaultTimeout)
```
Comment (jradtilbrook) [inline buildkite/resource_pipeline_template.go:215]:
```suggestion
	timeout, diags := pt.client.timeouts.Update(ctx, DefaultTimeout)
```

**Review by jradtilbrook — CHANGES_REQUESTED** [review]

File: buildkite/resource_pipeline_template.go (hunk 4 above)
Comment (james2791) [reply buildkite/resource_pipeline_template.go]: 👍👀

**Review by jradtilbrook — APPROVED** [review]

**Pull request merged**